snapcraft.yaml

```diff
@@ -62,3 +65,16 @@ parts:
       # Create a symbolic link to /usr/lib/<arch> where libfakeroot will live
       cd $SNAPCRAFT_PART_INSTALL/usr/lib/
       ln -s ${SNAPCRAFT_ARCH_TRIPLET} lib-arch
+
+  # When building a Classic image using a gadget snap which writes files to a bare partition, mcopy is used.
+  # There is a regression in mtools 4.0.32 which results in mcopy failing just
+  # in case the target directory does not already exist. 4.0.32 is available in
+  # 22.04 and 4.0.43 is available in 23.10, so when the core24 base is released
+  # mtools can be readded as a stage-package.
+  mtools:
```

Comment:
I must say I'm do not want to take this change. If there is a regression in mtools, I would prefer for the regression to be addressed as an SRU and updated properly in the archive instead. ubuntu-image is a tool which is used for building official Ubuntu image, so it *should not* use self-compiled tooling from outside of the archive.

Reply:
That's fair. I'll try to investigate what patch specifically resolves the issue - there's a post in the mailing list but I can't find a patch or any hosted VCS; I guess I have to dig into the release tarballs.
Once I've got it I'll try to initiative the SRU for it, however that works. Might be slow, I'm at the RISC-V Summit this week.

Reply:
mtools is actually more frustrating than I thought it would be, but conveniently the change between 4.0.33 and 4.0.34 ([the release cut to fix this regression](https://lists.gnu.org/archive/html/info-mtools/2021-07/msg00006.html)) is just:

```
dilyn@Ares:~/Downloads -> busybox diff mtools-4.0.33/mcopy.c mtools-4.0.34/mcopy.c
--- mtools-4.0.33/mcopy.c
+++ mtools-4.0.34/mcopy.c
@@ -107,7 +107,6 @@
 	Stream_t *File=mp->File;
 	Stream_t *Target, *Source;
 	struct MT_STAT stbuf;
-	ssize_t ret;
 	char errmsg[80];
 
 	File->Class->get_data(File, &mtime, 0, 0, 0);
@@ -164,7 +163,7 @@
 	if ((Target = SimpleFileOpen(0, 0, unixFile,
 				    O_WRONLY | O_CREAT | O_TRUNC,
 				    errmsg, 0, 0, 0))) {
-		ret = 0;
+		mt_off_t ret = 0;
 		if(needfilter && arg->textmode){
 			Source = open_filter(COPY(File),arg->convertCharset);
 			if (!Source)
```

Will test...

Reply:
This is actually a funny joke -- it's more complicated, because the Jammy package is actually 4.0.32. Harder to identify what small change is required.

Reply:
Update on this outstanding PR (sorry!), I've [filed a bug](https://bugs.launchpad.net/ubuntu/+source/mtools/+bug/2052731) against the mtools package and have started (I think...) the SRU process.